src/index.ts

```diff
@@ -327,6 +342,83 @@ export class FlashbotsBundleProvider extends providers.JsonRpcProvider {
     return this.sendRawBundle(signedTransactions, targetBlockNumber, opts)
   }
 
+  public async sendPrivateTransaction(
+    transaction: FlashbotsBundleTransaction,
+    opts?: {
+      maxBlockNumber?: number,
+      minTimestamp?: number,
+    },
+  ): Promise<FlashbotsPrivateTransaction> {
+    return this.sendPrivateRawTransaction(
+      await transaction.signer.signTransaction(transaction.transaction),
+      opts,
+    )
+  }
+
+  public async sendPrivateRawTransaction(
+    signedTransaction: string,
+    opts?: {
+      maxBlockNumber?: number,  // highest block # in which the bundle might be included
+      minTimestamp?: number,    // sim timestamp
+    },
+  ): Promise<FlashbotsPrivateTransaction> {
+    const params = {
+      tx: signedTransaction,
+      maxBlockNumber: opts?.maxBlockNumber,
+    }
+    const request = JSON.stringify(this.prepareRelayRequest('eth_sendPrivateTransaction', [params]))
+    const response = await this.request(request)
+    if (response.error !== undefined && response.error !== null) {
+      return {
+        error: {
+          message: response.error.message,
+          code: response.error.code
+        }
+      }
+    }
+
+    const transactionDetails = ethers.utils.parseTransaction(signedTransaction)
+    const privateTransaction = {
+      signedTransaction: signedTransaction,
+      hash: ethers.utils.keccak256(signedTransaction),
+      account: transactionDetails.from || "0x0",
+      nonce: transactionDetails.nonce,
+    }
+
+    const startBlockNumber = await this.genericProvider.getBlockNumber();
```

Comment:
this could be started before sending to relay, and awaited here

Reply:
:brain: good look, done